File: ABNLookupSampleCodeVBDotNet/ABRXMLSearchClient/ProxyXMLSearch.vb
Imports ABRXMLSearchClient.WebReference
'Imports ABRXMLSearchClient.au.gov.business.abr

' Strongly Typed Search - Uses the web service proxy classes to search ABRpublic. 
Public Class ProxyXMLSearch
   '-----------------------------------------------------------------------------------------------
   ' Search by ABN
   '-----------------------------------------------------------------------------------------------
    Public Function ABNSearch(ByVal searchString As String, _
                              ByVal history As String, _
                              ByVal guid As String) As WebReference.Payload

        Dim ABRSearch As WebReference.ABRXMLSearch = New WebReference.ABRXMLSearch
        Return ABRSearch.ABRSearchByABN(searchString, history, guid)
    End Function
    '-----------------------------------------------------------------------------------------------
    ' Search by ASIC
    '-----------------------------------------------------------------------------------------------
    Public Function ASICSearch(ByVal searchString As String, _
                              ByVal history As String, _
                              ByVal guid As String) As WebReference.Payload

        Dim ABRSearch As WebReference.ABRXMLSearch = New WebReference.ABRXMLSearch
        Return ABRSearch.ABRSearchByASIC(searchString, history, guid)

    End Function
    '-----------------------------------------------------------------------------------------------
    ' Search by Name
    '-----------------------------------------------------------------------------------------------
    Public Function NameSearch(ByVal guid As String, _
                               ByVal criteria As ExternalRequestNameSearch) As WebReference.Payload

        Dim ABRSearch As WebReference.ABRXMLSearch = New WebReference.ABRXMLSearch
        Return ABRSearch.ABRSearchByName(criteria, guid)
    End Function

End Class

File: ABNLookupSampleCodeCSharp35/SampleAbnLookupClient/ProxyXmlSearch.cs
using System;
using AbnLookup.SearchClientCSharpe.ServiceReferenceAbnLookup;

namespace AbnLookup.SearchClientCSharpe {
   class ProxyXmlSearch {
      //  Private constructor
      private ProxyXmlSearch() { }
      // -----------------------------------------------------------------------------------------------
      //  Search by ABN
      // -----------------------------------------------------------------------------------------------
      public static Payload AbnSearch(string searchText, string history, string guid) {
         ABRXMLSearchSoapClient Search = new ABRXMLSearchSoapClient();
         return Search.ABRSearchByABN(searchText, history, guid);
      }
      // -----------------------------------------------------------------------------------------------
      //  Search by ASIC
      // -----------------------------------------------------------------------------------------------
      public static Payload AsicSearch(string searchText, string history, string guid) {
         ABRXMLSearchSoapClient Search = new ABRXMLSearchSoapClient();
         return Search.ABRSearchByASIC(searchText, history, guid);
      }
      // -----------------------------------------------------------------------------------------------
      //  Search by Name
      // -----------------------------------------------------------------------------------------------
      public static Payload NameSearch(string guid, ExternalRequestNameSearch criteria) {
         ABRXMLSearchSoapClient Search = new ABRXMLSearchSoapClient();
         return Search.ABRSearchByName(criteria, guid);
      }
      // -----------------------------------------------------------------------------------------------
      //  Search by Postcode
      // -----------------------------------------------------------------------------------------------
      public static Payload PostcodeSearch(string guid, string postcode) {
         ABRXMLSearchSoapClient Search = new ABRXMLSearchSoapClient();
         return Search.SearchByPostcode(postcode, guid);
      }
   }
}

